src/Export/Errors/ExportErrors.php

```diff
@@ -0,0 +1,85 @@
+<?php
+
+declare(strict_types=1);
+
+namespace FINDOLOGIC\FinSearch\Export\Errors;
+
+class ExportErrors
+{
+    /** @var string[] */
+    private $general = [];
+
+    /** @var ProductError[] */
+    private $products = [];
+
+    public function hasErrors(): bool
+    {
+        if (!empty($this->general)) {
+            return true;
+        }
+
+        foreach ($this->products as $product) {
+            if (!empty($product->getErrors())) {
+                return true;
+            }
+        }
+
+        return false;
+    }
+
+    public function buildErrorResponse(): array
+    {
+        $productErrors = array_map(function (ProductError $product) {
+            return $product->toArray();
+        }, $this->products);
+
+        return [
+            'general' => $this->general,
+            'products' => array_values($productErrors),
```

Comment:
Nothing is merged here anyway. The `toArray` method is used, so we get an array representation of the objects, which we use for the JSON later on. We use `array_values` only to get rid of the id of the products as key, since in the JSON, we already print out the id and the errors anyway.